kie-server-parent/kie-server-services/kie-server-services-openshift/src/main/java/org/kie/server/services/openshift/impl/OpenShiftStartupStrategy.java

```diff
@@ -68,17 +68,14 @@ public WatchRunner(String serverId) {
         public void run() {
             try (OpenShiftClient client = clouldClientHelper.get()) {
                 logger.info("Watching ConfigMap in namespace: [{}]", client.getNamespace());
-                try (Watch watchable = client.configMaps().watch(new Watcher<ConfigMap>() {
+                try (Watch watchable = client.configMaps().withName(kieServerId).watch(new Watcher<ConfigMap>() {
                     @Override
                     public void eventReceived(Action action, ConfigMap kieServerState) {
                         logger.debug("Event - Action: {}, {} on ConfigMap ",
                                     action, kieServerState.getMetadata().getName());
 
                         DeploymentConfig dc = client.deploymentConfigs().withName(kieServerId).get();
```

Comment:
This is a potential source of inconsistency as the code expects that deployment config name is equal to Kie server id.
This is a default configuration in our templates, but Kie server id environment variable can be modified later or customer can create own templates which don't follow this precondition.
@rhtevan Wouldn't it be better to adjust code to filter all deployment configs based on environment property KIE_SERVER_ID ?

Reply:
@sutaakar 
A simplest way of maintaining the key relationships between ServerTemplate, KieServer, DeploymentConfig and ConfigMap, is to align KieServerId with the names of DC and CM. (We tends to align the concepts of KieServer with DeploymentConfig rather than Pods). 
As discovered, the default value of KieServerId is indirectly dictated by DC name (See https://github.com/jboss-container-images/jboss-kie-modules/blob/65371750d322288ca81b128ce1d32b190bdf0356/jboss-kie-kieserver/added/launch/jboss-kie-kieserver.sh#L451-L470) and loaded through environment variable and java system property. (See https://github.com/rhtevan/droolsjbpm-integration/blob/5be8e945e9df3b68609a717ddf62fac5c0985ead/kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/model/KieServerConfigItem.java#L55; and   https://github.com/rhtevan/droolsjbpm-integration/blob/5be8e945e9df3b68609a717ddf62fac5c0985ead/kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/KieServerEnvironment.java#L28)

Therefore, that alignment has already been somewhat protected. Having said that, nothing stops end user to deliberately set the KIE_SERVER_ID environment variable different from the DC name right now. As a remedy, we could either discourage or ignore such env settings by:

- Clearly document it. 
- Remove it from Out-Of-The-Box template
- Ignore such env variable and always use the value generated by the initialization scripts from the standard rhpam image.

Reply:
@errantepiphany @sutaakar 
David, It's suggested to loop you into this discussion, and see if you are agree and or having some other thoughts. In nut shell, it is important to have a consistent naming between DC, CM, KieServer Pod and KieServerId. Right now, it is somewhat enforced and protected by Template parameter and KieServer initialization scripts. Having said that, it does not completely prevent from misuse. To mitigate such risk, we could start with documentation and move on to, perhaps, constrain the capability of arbitrary configuration of KieServerId by adjusting Template and iniit scripts.  What do you think?

Reply:
@rhtevan Sorry for the late reply. I'm looking... (and we have a call in 5 minutes too). ;)

Reply:
@sutaakar @errantepiphany 
Based on the discussion, it has been agreed upon to provide documentation to mitigate such risk from the coming 7.3.0 release, and log a separate JIRA (https://issues.jboss.org/browse/JBPM-8150) for future enhancement.